dali/operators/geometry/affine_transforms/transform_scale.cc

```diff
@@ -78,16 +78,15 @@ class TransformScaleCPU
   }
 
   void ProcessArgs(const OpSpec &spec, const workspace_t<CPUBackend> &ws) {
-    int repeat = IsConstantTransform() ? 0 : nsamples_;
     assert(scale_.IsDefined());
-    scale_.Read(spec, ws, repeat);
-    ndim_ = scale_[0].size();
+    scale_.Acquire(spec, ws, nsamples_, true);
```

Comment:
Nitpick: If there's an input, you could use expected shape in order to broadcast uniform scale.